bears/js/JSONFormatBear.py

```diff
@@ -48,8 +26,29 @@ def run(self,
         :param tab_width:    Number of spaces to indent.
         :param keep_unicode: Wether or not to escape unicode values using ASCII.
         """
-        return self.lint(filename,
-                         file,
-                         sort_keys=json_sort,
-                         indent=tab_width,
-                         ensure_ascii=not keep_unicode)
+        try:
+            json_content = json.loads(''.join(file),
+                                      object_pairs_hook=OrderedDict)
+        except self.DecodeError as err:
+            yield Result.from_values(
+                self,
+                "This file does not contain parsable JSON. " + repr(str(err)),
+                file=filename)
+            return
+
+        corrected = json.dumps(json_content,
+                               sort_keys=json_sort,
+                               indent=tab_width,
+                               ensure_ascii=not keep_unicode).splitlines(True)
+        # Because of a bug in several python versions we have to correct
+        # whitespace here.
+        corrected = [line.rstrip(" \n") + "\n" for line in corrected]
+
+        diffs = Diff.from_string_arrays(file, corrected).split_diff()
```

Comment:
let's not split this one, if keys are sorted, applying only one of the splitted diffs may result in wrong stuff. Be sure to use the ranges of the splitted diff though